### ISSUE TITLE
API

### ISSUE THREAD

**Issue by shmidt** [EyreFree/EFQRCode#4]
Thanks for sharing.

My suggestion is to make API a little bit more readable by adding couple of classes: `Icon` and `Watermark`.

Then, instead:

```
EFQRCode.generate(
    content: String, 
    inputCorrectionLevel: EFInputCorrectionLevel, 
    size: CGFloat, 
    magnification: UInt?, 
    backgroundColor: UIColor, 
    foregroundColor: UIColor, 
    icon: UIImage?, 
    iconSize: CGFloat?, 
    isIconColorful: Bool, 
    watermark: UIImage?, 
    watermarkMode: EFWatermarkMode, 
    isWatermarkColorful: Bool
)
```

It will be
```
EFQRCode.generate(
    content: String, 
    correction: EFInputCorrectionLevel, 
    size: CGFloat, 
    magnification: UInt?, 
    backgroundColor: UIColor, 
    foregroundColor: UIColor, 
    icon: Icon(image: UIImage, size: CGFloat, colorful: Bool)?, 
    watermark: Watermark(image: UIImage, mode: EFWatermarkMode, colorful: Bool)?
)
```

**Comment by EyreFree** [comment]
Oh, this is a good idea, I will think about this and add this in the future, maybe next release, thanks, 👍🏻

**Comment by EyreFree** [comment]
Fixed in [1.2.1](https://github.com/EyreFree/EFQRCode/pull/6).
